src/metabase/cmd/rotate_encryption_key.clj

```diff
@@ -0,0 +1,35 @@
+(ns metabase.cmd.rotate-encryption-key
+  (:require [cheshire.core :as json]
+            [clojure.java.jdbc :as jdbc]
+            [metabase.db :as mdb]
+            [metabase.db.connection :as mdb.conn]
+            [metabase.models :refer [Database Setting]]
+            [metabase.util.encryption :as encrypt]
+            [toucan.db :as db]))
+
+(defn rotate-encryption-key!
+  "Rotate the current configured db using the current MB_ENCRYPTION_SECRET_KEY env var and `to-key` argument."
+  [to-key]
+  (mdb/setup-db!)
+  (try
+    (let [encrypt-fn (if to-key
+                       (partial encrypt/maybe-encrypt
+                                (encrypt/validate-and-hash-secret-key to-key))
+                       identity)]
+      (jdbc/with-db-transaction [t-conn (mdb.conn/jdbc-spec)]
+        (doseq [[key value] (db/select-field->field :key :value Setting)]
+          (when (encrypt/possibly-encrypted-string? value)
+            (throw  (Exception. "MB_ENCRYPTION_SECRET_KEY does not correcty decrypt files")))
+          (jdbc/update! t-conn
+                        :setting
+                        {:value (encrypt-fn value)}
+                        ["setting.key = ?" key]))
+        (doseq [[id details] (db/select-id->field :details Database)]
+          (jdbc/update! t-conn
+                        :metabase_database
+                        {:details (encrypt-fn (json/encode details))}
+                        ["metabase_database.id = ?" id])
+          (when (encrypt/possibly-encrypted-string? details)
+            (throw  (Exception. "MB_ENCRYPTION_SECRET_KEY does not correcty decrypt files")))))
+      true)
+    (catch Throwable e (println "MB_ENCRYPTION_SECRET_KEY does not correcty decrypt the existing data"))))
```

Comment:
Instead of println you should use `log/error`, `log/debug`, `log/info`, etc. in your commands. It lets us turn off the logging in tests for example